### ISSUE TITLE
Custom labels on xAxis <BarChart/>

### ISSUE THREAD

**Issue by audn** [Abhinandan-Kushwaha/react-native-gifted-charts#422]
Hi,
I couldn't find anything in the docs on how to manipulate the xAxis label to get the same results as the picture below.
Can anyone point me in the right direction?

Current:
<img width="333" alt="image" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/21041399/efbfbb3d-d3e7-48bf-b8a1-b2907b007a25">
Wanted:
<img width="716" alt="image" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/21041399/3d37c10b-a285-4b8e-ba48-364ed0488970">

I tried messign with `width` on the `xAxisLabelTextStyle`, but it wont center underneath the bar.

Code:
````tsx
const App = () => {
const data = [
      {
         date: '2023-10-31',
         stacks: [
            { value: 244.2, color: '#BD6130', label: 'Protein' },
            { value: 532.8, color: '#63B979', label: 'Carbohydrates' },
            { value: 499.5, color: '#C0AD69', label: 'Fat' },
         ],
      },
      {
         date: '2023-10-30',
         stacks: [
            { value: 75.60000000000001, color: '#BD6130', label: 'Protein' },
            {
               value: 533.8000000000001,
               color: '#63B979',
               label: 'Carbohydrates',
            },
            { value: 63, color: '#C0AD69', label: 'Fat' },
         ],
      },
      {
         date: '2023-10-29',
         stacks: [
            { value: 75.60000000000001, color: '#BD6130', label: 'Protein' },
            {
               value: 533.8000000000001,
               color: '#63B979',
               label: 'Carbohydrates',
            },
            { value: 63, color: '#C0AD69', label: 'Fat' },
         ],
      },
      {
         date: '2023-10-10',
         stacks: [
            { value: 0, color: '#BD6130', label: 'Protein' },
            { value: 144.96, color: '#63B979', label: 'Carbohydrates' },
            { value: 0, color: '#C0AD69', label: 'Fat' },
         ],
      },
   ];

  return (
     <BarChart
        maxValue={Math.round(user?.healthProfile?.targetCalories / 1000) * 1000}
        initialSpacing={0}
        stepHeight={120}
        width={Dimensions.get('screen').width}
        barWidth={12}
        rulesColor={tw.color('bg-app-400')}
        noOfSections={4}
        spacing={5}
        barBorderRadius={0}
        yAxisColor={'transparent'}
        xAxisColor={tw.color('bg-app-300')}
        // yAxisSide={yAxisSides.RIGHT}
        yAxisTextStyle={twStyle('text-on-200')}
        xAxisLabelTextStyle={twStyle(
           'bg-red-500 mx-auto w-full overflow-auto text-white z-50 text-center '
        )}
        stackData={
           !isLoading
              ? data.map((x, index) => {
                   const label =
                      index === 0 || (index + 1) % 5 === 0
                         ? x.date
                         : '';
                   return {
                      ...x,
                      label,
                   };
                })
              : []
        }
     />
  );
}```

**Comment by Abhinandan-Kushwaha** [comment]
Hi @audn 👋
Try giving a negative marginLeft inside `xAxisLabelTextStyle`, something like `marginLeft: -20`

A negative margin is needed because the x-axis label starts with the starting point of bar. Since the label text is longer than barWidth, we need to shift the label towards the left.

**Comment by audn** [comment]
Will try later! Any ideas for the white bar over the label though?

**Comment by Abhinandan-Kushwaha** [comment]
Hi @audn 👋
Yes, we can display a white bar over the labels. In fact, we can use any custom UI component as x-axis label using the property `labelComponent` instead of `label`.

Here's an example-

```js
const daata = [
  {
    date: '2023-10-31',
    stacks: [
      {value: 244.2, color: '#BD6130', label: 'Protein'},
      {value: 532.8, color: '#63B979', label: 'Carbohydrates'},
      {value: 499.5, color: '#C0AD69', label: 'Fat'},
    ],
  },
  {
    date: '2023-10-30',
    stacks: [
      {value: 75.60000000000001, color: '#BD6130', label: 'Protein'},
      {
        value: 533.8000000000001,
        color: '#63B979',
        label: 'Carbohydrates',
      },
      {value: 63, color: '#C0AD69', label: 'Fat'},
    ],
  },
  {
    date: '2023-10-29',
    stacks: [
      {value: 75.60000000000001, color: '#BD6130', label: 'Protein'},
      {
        value: 533.8000000000001,
        color: '#63B979',
        label: 'Carbohydrates',
      },
      {value: 63, color: '#C0AD69', label: 'Fat'},
    ],
  },
  {
    date: '2023-10-10',
    stacks: [
      {value: 0, color: '#BD6130', label: 'Protein'},
      {value: 144.96, color: '#63B979', label: 'Carbohydrates'},
      {value: 0, color: '#C0AD69', label: 'Fat'},
    ],
  },
];

const getLabelComponent = label => {
  return (
    <View
      style={{
        justifyContent: 'center',
        alignItems: 'center',
        height: 65,
        width: 50,
        position: 'absolute',
        top: -30,
        left: -20,
      }}>
      <View
        style={{
          width: 3,
          height: 15,
          backgroundColor: 'white',
          marginBottom: 8,
        }}
      />
      <Text style={{color: 'white', fontWeight: 'bold'}}>
        {'Oct ' + label.substring(label.length - 2)}
      </Text>
    </View>
  );
};

return (
  <View style={{flex: 1, backgroundColor: '#333'}}>
    <BarChart
      initialSpacing={0}
      stepHeight={120}
      width={Dimensions.get('screen').width}
      barWidth={12}
      noOfSections={4}
      spacing={5}
      barBorderRadius={0}
      yAxisColor={'transparent'}
      xAxisColor={'lightgray'}
      yAxisTextStyle={{color: 'lightgray'}}
      xAxisLabelTextStyle={{
        color: 'white',
        width: 100,
      }}
      stackData={[...daata, ...daata, ...daata].map((x, index) => {
        return {
          ...x,
          labelComponent: () => {
            if (index === 0 || (index + 1) % 5 === 0) {
              return getLabelComponent(x.date);
            }
            return null;
          },
        };
      })}
    />
  </View>
);
```

The output is-

<img width="365" alt="Screenshot 2023-10-31 at 1 08 41 PM" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/20596944/b52e6af7-8a87-4de8-89f6-741c14b93644">

I hope it helps.

**Comment by audn** [comment]
Thank you for the quick reply!
The only problem now is that the bar is growing outside the scrollview when I have my `stepHeight` at `120`.
<img width="419" alt="image" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/21041399/8cca18c9-8401-4c56-8a4e-6a58cb02b475">
I'm able to fix it by lowering the `stepHeight` and increasing `noOfSections`, but thats not what I want to do.
Do you know a workaround?

**Comment by Abhinandan-Kushwaha** [comment]
In my comment above I have used `stepHeight={120}` and the bars don't go outside the ScrollView.

<img width="368" alt="Screenshot 2023-11-01 at 10 45 48 AM" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/20596944/ebd1365d-4eb8-4571-b707-fb7797feff02">

Can you check which prop is causing this behaviour? Try commenting out all the props and then enabling them one by one.

**Comment by audn** [comment]
I think you're adding it on the wrong element. I'm adding the color on the parent. See how your bg color doesn't affect the YAxisLabel

**Comment by Abhinandan-Kushwaha** [comment]
Got it @audn 
Will fix it later sometime. Till then you can give some `paddingBottom` to the surrounding View to fix it.

**Comment by Abhinandan-Kushwaha** [comment]
Thanks @audn for pointing it out. It has been now fixed in version `1.3.16`🎉
